rplugin/python3/denite/ui/bridge.py

```diff
@@ -0,0 +1,54 @@
+import weakref
+from .default import Default
+from ..prompt.prompt import Prompt
+from ..prompt.context import Context
+
+
+class Bridge(Default):
+    def __init__(self, vim):
+        super().__init__(vim)
+        self.__prompt = BridgePrompt(vim, weakref.proxy(self))
+
+    def input_loop(self):
+        self.__prompt.start()
+
+
+class BridgeContext(Context):
+    __slots__ = ('caret_locus', '_context')
```

Comment:
Saving an entire `_context` is a bit too much. Context should hold a minimum information to restore the status to reduce the memory usage.